### ISSUE TITLE
Build failure on aarch64 with GCC11

### ISSUE THREAD

**Issue by ggardet** [sherpa/sherpa#1186]
With GCC11, Sherpa 4.13.1 fails on aarch64 with:
```
[  174s] CCompilerOpt.dist_test[576] : CCompilerOpt._dist_test_spawn[711] : Command (gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -mbranch-protection=standard -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -DOPENSSL_LOAD_CONF -fwrapv -fno-semantic-interposition -mbranch-protection=standard -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -IVendor/ -mbranch-protection=standard -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -IVendor/ -fPIC -I/usr/include/python3.9 -c /usr/lib64/python3.9/site-packages/numpy/distutils/checks/cpu_asimdfhm.c -o /tmp/tmp0x2ogkio/usr/lib64/python3.9/site-packages/numpy/distutils/checks/cpu_asimdfhm.o -MMD -MF /tmp/tmp0x2ogkio/usr/lib64/python3.9/site-packages/numpy/distutils/checks/cpu_asimdfhm.o.d -march=armv8.2-a+fp16+fp16fml -Werror) failed with exit status 1 output -> 
[  174s] /usr/lib64/python3.9/site-packages/numpy/distutils/checks/cpu_asimdfhm.c: In function ‘main’:
[  174s] /usr/lib64/python3.9/site-packages/numpy/distutils/checks/cpu_asimdfhm.c:13:35: error: implicit declaration of function ‘vfmlal_low_u32’; did you mean ‘vfmlal_low_f16’? [-Werror=implicit-function-declaration]
[  174s]    13 |     int ret  = (int)vget_lane_f32(vfmlal_low_u32(vlf, vlhp, vlhp), 0);
[  174s]       |                                   ^~~~~~~~~~~~~~
[  174s]       |                                   vfmlal_low_f16
[  174s] /usr/lib64/python3.9/site-packages/numpy/distutils/checks/cpu_asimdfhm.c:13:35: error: incompatible type for argument 1 of ‘vget_lane_f32’
[  174s]    13 |     int ret  = (int)vget_lane_f32(vfmlal_low_u32(vlf, vlhp, vlhp), 0);
[  174s]       |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[  174s]       |                                   |
[  174s]       |                                   int
[  174s] In file included from /usr/lib64/python3.9/site-packages/numpy/distutils/checks/cpu_asimdfhm.c:4:
[  174s] /usr/lib64/gcc/aarch64-suse-linux/11/include/arm_neon.h:3169:28: note: expected ‘float32x2_t’ but argument is of type ‘int’
[  174s]  3169 | vget_lane_f32 (float32x2_t __a, const int __b)
[  174s]       |                ~~~~~~~~~~~~^~~
[  174s] /usr/lib64/python3.9/site-packages/numpy/distutils/checks/cpu_asimdfhm.c:14:36: error: implicit declaration of function ‘vfmlslq_high_u32’; did you mean ‘vmlsl_high_u32’? [-Werror=implicit-function-declaration]
[  174s]    14 |         ret += (int)vgetq_lane_f32(vfmlslq_high_u32(vf, vhp, vhp), 0);
[  174s]       |                                    ^~~~~~~~~~~~~~~~
[  174s]       |                                    vmlsl_high_u32
[  174s] /usr/lib64/python3.9/site-packages/numpy/distutils/checks/cpu_asimdfhm.c:14:36: error: incompatible type for argument 1 of ‘vgetq_lane_f32’
[  174s]    14 |         ret += (int)vgetq_lane_f32(vfmlslq_high_u32(vf, vhp, vhp), 0);
[  174s]       |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[  174s]       |                                    |
[  174s]       |                                    int
[  174s] In file included from /usr/lib64/python3.9/site-packages/numpy/distutils/checks/cpu_asimdfhm.c:4:
[  174s] /usr/lib64/gcc/aarch64-suse-linux/11/include/arm_neon.h:3269:29: note: expected ‘float32x4_t’ but argument is of type ‘int’
[  174s]  3269 | vgetq_lane_f32 (float32x4_t __a, const int __b)
[  174s]       |                 ~~~~~~~~~~~~^~~
[  174s] cc1: all warnings being treated as errors
```

And this one which also affects ppc64/ppc64le:
```
[  177s] sherpa/utils/src/cephes/const.c:94:1: error: function ‘__builtin_inff’ is initialized like a variable
[  177s]    94 | double INFINITY =  1.79769313486231570815E308;    /* 2**1024*(1-MACHEP) */
[  177s]       | ^~~~~~
[  177s] In file included from sherpa/utils/src/cephes/mconf.h:74,
[  177s]                  from sherpa/utils/src/cephes/const.c:62:
[  177s] sherpa/utils/src/cephes/const.c:99:8: error: expected declaration specifiers or ‘...’ before string constant
[  177s]    99 | double NAN = 0.0;
[  177s]       |        ^~~
```

**Comment by DougBurke** [comment]
The first points look to be from numpy and not directly from Sherpa: I don't know if we're missing some setup/specification or maybe we need a bump to the minimum-supported numpy?

The later ones are definite sherpa issues!

**Comment by DougBurke** [comment]
https://stackoverflow.com/questions/35534478/including-algorithm-and-limits-causes-invalid-pure-specifier-compilation-e / https://en.cppreference.com/w/cpp/numeric/math/INFINITY suggest we shouldn't be defining the `INFINITY` symbol, but I don't see a lot of chatter about problems with CEPHES code and this - which is where this code comes from.

I note that SciPy doesn't define `INFINITY` in its copy of cephes - see https://github.com/scipy/scipy/blob/master/scipy/special/cephes/const.c - and for the one file we use INFINITY - which is `gamma.c` the SciPy code uses `NPY_INFINITE` - although the code is rather different to ours:

https://github.com/scipy/scipy/blob/master/scipy/special/cephes/gamma.c

vs

https://github.com/sherpa/sherpa/blob/main/sherpa/utils/src/cephes/gamma.c

**Comment by DougBurke** [comment]
If we required scipy then we could just drop our cephes code ;-)

Although I haven't checked that they provide everything we need, but I imagine they do...

**Comment by hamogu** [comment]
Anytime we can reduce the amount of C code that we need to compile ourselves, we make our live simpler.

**Comment by ggardet** [comment]
This is error is fixed with this numpy commit: https://github.com/numpy/numpy/commit/2113cad8cbb4c4bc9469e0c94025bd7cbbe6f2d6

**Comment by ggardet** [comment]
> But this one remains:
> 
> ```
> [  177s] sherpa/utils/src/cephes/const.c:94:1: error: function ‘__builtin_inff’ is initialized like a variable
> [  177s]    94 | double INFINITY =  1.79769313486231570815E308;    /* 2**1024*(1-MACHEP) */
> [  177s]       | ^~~~~~
> [  177s] In file included from sherpa/utils/src/cephes/mconf.h:74,
> [  177s]                  from sherpa/utils/src/cephes/const.c:62:
> [  177s] sherpa/utils/src/cephes/const.c:99:8: error: expected declaration specifiers or ‘...’ before string constant
> [  177s]    99 | double NAN = 0.0;
> [  177s]       |        ^~~
> ```

This is fixed in version `4.14.0`.

**Comment by DougBurke** [comment]
Ta for checking, @ggardet
